concrete-boolean/src/client_key/mod.rs

```diff
@@ -21,11 +18,45 @@ use serde::{Deserialize, Serialize};
 /// * `glwe_secret_key` - a GLWE secret key, used to generate the bootstrapping keys and key
 /// switching keys.
 /// * `parameters` - the cryptographic parameter set.
-#[derive(Serialize, Clone, Deserialize, PartialEq, Debug)]
+#[derive(Serialize, Deserialize)]
 pub struct ClientKey {
-    pub(crate) lwe_secret_key: LweSecretKey<BinaryKeyKind, Vec<u32>>,
-    pub(crate) glwe_secret_key: GlweSecretKey<BinaryKeyKind, Vec<u32>>,
+    pub(crate) lwe_secret_key: LweSecretKey32,
+    pub(crate) glwe_secret_key: GlweSecretKey32,
     pub(crate) parameters: BooleanParameters,
+    #[serde(skip, default = "crate::default_engine")]
```

Comment:
Here why do you need to skip this serialization and specify a default instead?

Reply:
The `CoreEngine` is not serializable so we have to skip it, and even if it was serializable there would be not point in serializing it what we care about is serializing the actual data of the key